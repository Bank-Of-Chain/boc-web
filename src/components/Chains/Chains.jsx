import React from "react"

// mterial-ui components
import { makeStyles } from "@material-ui/core/styles"
import Tooltip from "@material-ui/core/Tooltip";

// === Utils === //
import map from 'lodash/map';

const styles = {
  img: {
    width: '2rem',
    height: '2rem',
    marginRight: '10px',
    maxWidth: '100%',
    padding: 2,
    verticalAlign: 'middle',
    background: '#fff',
    cursor: 'pointer',
    border: '1px solid #000',
  }
}

const useStyles = makeStyles(styles)

export default function Chains (props) {
  const { array, handleClick } = props
  console.log('array=', array);
  const classes = useStyles()
  return (
    <div>
      {map(array, item => (
        <Tooltip
          key={item.chainId}
          title={item.name}
          placement={window.innerWidth > 959 ? "top" : "left"}
          classes={{ tooltip: classes.tooltip }}
        >
          <a onClick={() => handleClick(item)}>
            <img className={classes.img} src={require(`./images/${item.chainId}.svg`)} alt='' />
          </a>
        </Tooltip>
      ))}
    </div>
  )
}

Chains.propTypes = {}
