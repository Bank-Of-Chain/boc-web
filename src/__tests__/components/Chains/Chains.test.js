import React from "react";
import { render } from "@testing-library/react";
import Chains from "../../../components/Chains/Chains";

test("Chains Component Render", () => {
  const { asFragment } = render(<Chains />);
  expect(asFragment()).toMatchSnapshot();
});
