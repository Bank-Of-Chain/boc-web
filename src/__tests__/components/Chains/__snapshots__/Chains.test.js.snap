// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chains Component Render 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
